torchao/csrc/cpu/quantized_sdpa.cpp

```diff
@@ -1873,30 +2500,63 @@ at::Tensor _qscaled_dot_product_cpu(
   TORCH_CHECK(!attn_mask.has_value() ||
           (attn_mask.value().dim() == 2 || attn_mask.value().dim() == 4),
     "_qscaled_dot_product_cpu: Attention mask dim in {2, 4}");
+  if (dtype == at::ScalarType::Float8_e4m3fn) {
+    TORCH_CHECK(q_zp == 0 && k_zp == 0 && v_zp == 0 && a_zp == 0 && o_zp == 0,
+      "_qscaled_dot_product_cpu: Don't accept zero point for Float8_e4m3");
+  }
 
-  #ifdef CPU_CAPABILITY_AVX512
-    if (at::native::cpublas::could_pack(dtype)) {
-        at::Tensor output = at::empty_like(query, query.options()).transpose(1, 2);
-        sdpa_int8_fused_kernel(output, query, key, value,
-            dropout_p, is_causal, attn_mask, scale,
-            q_scale, q_zp,
-            k_scale, k_zp,
-            v_scale, v_zp,
-            a_scale, a_zp,
-            o_scale, o_zp);
-        return output.transpose(1, 2);
-    } else {
-  #endif // CPU_CAPABILITY_AVX512
-        return sdpa_int8_math_kernel(query, key, value,
+  if (dtype == at::ScalarType::Byte) {
+#ifdef CPU_CAPABILITY_AVX512
+      if (at::native::cpublas::could_pack(dtype)) {
```

Comment:
Do we always need pack on supported platforms ? Is there any cases where do packing is slower than plain format ?

Reply:
Thanks for your suggestion!
For the cases we care about, for example landing zone models, we have confirmed that packing is better.
For the general cases, we can tune for need_pack in the future.